src/utils/rast_seed_mapper.py

```diff
@@ -0,0 +1,314 @@
+            raw_id: Raw ID string from ontology
+            
+        Returns:
+            Cleaned seed.role ID or None if parsing fails
+        """
+        if not raw_id:
+            return None
+            
+        # URL format with Role parameter
+        if "Role=" in raw_id:
+            try:
+                role_number = raw_id.split("Role=")[-1]
+                return f"seed.role:{role_number}"
+            except IndexError:
+                logger.warning(f"Failed to parse role number from URL: {raw_id}")
+                return None
+                
+        # Already in clean format
+        if raw_id.startswith("seed.role:"):
+            return raw_id
+            
+        # Try generic parsing for OBO-style IDs (e.g., seed.role_0000000001234)
+        try:
+            ontology_part = raw_id.split('/')[-1]
+            # Only convert underscore if it looks like an OBO ID (namespace_number)
+            if '_' in ontology_part and ontology_part.startswith('seed.role_'):
+                return ontology_part.replace("_", ":", 1)
+        except Exception as e:
+            logger.warning(f"Failed to parse ID: {raw_id}, error: {e}")
+            
+        return None
+    
+    def split_multi_function(self, annotation: str) -> List[str]:
+        """
+        Split multi-function annotations into individual components.
+        
+        Args:
+            annotation: RAST annotation string
+            
+        Returns:
+            List of individual function annotations
+        """
+        if not annotation:
+            return []
+            
+        # Start with the full annotation
+        parts = [annotation]
+        
+        # Split by each separator in order
+        for separator in self.multi_func_separators:
+            new_parts = []
+            for part in parts:
+                split_parts = part.split(separator)
+                new_parts.extend(p.strip() for p in split_parts if p.strip())
+            parts = new_parts
+            
+        return parts
+    
+    def map_annotation(self, annotation: str) -> Optional[str]:
+        """
+        Map a RAST annotation to its SEED role ID.
+        
+        Tries multiple strategies:
+        1. Direct exact match
+        2. Individual parts of multi-function annotations
+        
+        Args:
+            annotation: RAST annotation string
+            
+        Returns:
+            seed.role ID if found, None otherwise
+        """
+        if not annotation:
+            return None
+            
+        # Try direct match first
+        if annotation in self.seed_mapping:
+            return self.seed_mapping[annotation]
+            
+        # Try splitting multi-function annotations
+        parts = self.split_multi_function(annotation)
+        
+        if len(parts) > 1:
+            # Try each part
+            for part in parts:
+                if part in self.seed_mapping:
+                    return self.seed_mapping[part]
```

Comment:
This will only return the first mapping that it comes across and can match. Do you want to return as many matches as possible or just one?